src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy

```diff
@@ -45,7 +47,12 @@ class Liberty implements Plugin<Project> {
         project.configurations.create('libertyLicense')
         project.configurations.create('libertyRuntime')
 
-        def currentServer = project.liberty.server
+        project.afterEvaluate{
+            if (project.liberty.server == null) {
+                project.liberty.server = copyProperties(project.liberty)
+            }
+            setServersForTasks(project)
```

Comment:
Add a comment before this method explaining why you have to set the server object on the tasks.